### ISSUE TITLE
Loading 8051 memory other than rom

### ISSUE THREAD

**Issue by kilate** [danieldrotos/ucsim#3]
For the 8051 sim, it doesn't seem possible to load xram or any  address space other than rom  (let's say a paged/banked rom) with the content of a file, via file or dn command. Or am I missing something?

**Comment by danieldrotos** [comment]
Yes, you are correct, this is not possible yet. I add it to the TODO list.

**Comment by kilate** [comment]
Hi,

thanks for replying and a big 'thank you' for working on and sharing this project with us.

As a contribution find below an example, that you can add to documentation if you want, on how to load a paged/banked rom. Maybe not the best way, but it works for my case.



- this is really big for a 8051 (32 pages x 49k).
```
memory create chip bigrom_chip 0x180000 8
```
- Only 4 extra (plus the main page0) pages/banks in use for now.
```
memory create addressspace page0 0x4000 0xc000
memory create addressspace page1 0x4000 0xc000
memory create addressspace page2 0x4000 0xc000
memory create addressspace page3 0x4000 0xc000
memory create addressspace page4 0x4000 0xc000
```

- Page 0 is part of the main code (64k) and it resides in the rom_chip
```
memory create addressdecoder page0 0x4000 0xffff rom_chip 0x4000
```

- other pages go to the bigrom_chip
```
memory create addressdecoder page1 0x4000 0xffff bigrom_chip 0x0
memory create addressdecoder page2 0x4000 0xffff bigrom_chip 0xc000
memory create addressdecoder page3 0x4000 0xffff bigrom_chip 0x18000
memory create addressdecoder page4 0x4000 0xffff bigrom_chip 0x24000
```

- SRF 0x90 (P1) first five bits are used to select the active page
```
memory create banker sfr 0x90 0x1f rom 0x4000 0xffff
memory create bank rom 0x4000 0 rom_chip 0xc000
memory create bank rom 0x4000 1 bigrom_chip 0x0
memory create bank rom 0x4000 2 bigrom_chip 0xc000
memory create bank rom 0x4000 3 bigrom_chip 0x18000
memory create bank rom 0x4000 4 bigrom_chip 0x24000
```

- Loading the code files to the rom. Select page0  and load and main file (64k) starting from 0x0
```
set memory sfr 0x90 0
file "p0.ihex"
```

- Select each extra page and load its file (49k each). IMPORTANT: each extra page file starts at 0x4000)
 - e.g. objcopy -O ihex -I binary --change-address=0x4000 p1_file.bin p1.ihex
```
set memory sfr 0x90 1
file "p1.ihex"
set memory sfr 0x90 2
file "p2.ihex"
set memory sfr 0x90 3
file "p3.ihex"
set memory sfr 0x90 4
file "p4.ihex"
```
- back to page0
```
set memory sfr 0x90 0
<fun starts here>
```

**Comment by danieldrotos** [comment]
Version 0.9.8 accept input specifier in place of filename, in the form:

`filename@memory:offset`

where `@memory` and `:offset` are optional of course. Content of the file is loaded into specified memory which can be address space or chip. Offset is added to address loded from file to calculate memory index.
